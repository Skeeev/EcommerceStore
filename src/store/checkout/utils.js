import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
