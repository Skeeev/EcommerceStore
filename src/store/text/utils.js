export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
