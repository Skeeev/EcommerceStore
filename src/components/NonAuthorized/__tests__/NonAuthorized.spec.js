import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { NonAuthorized } from '..';

describe(`${getComponentDisplayName(NonAuthorized)} component`, () => {
  const renderComponent = props => shallow(<NonAuthorized {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      navigate: jest.fn()
    });

    expect(component).toMatchSnapshot();
  });
});
