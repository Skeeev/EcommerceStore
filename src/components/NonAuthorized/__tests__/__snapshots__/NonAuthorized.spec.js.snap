// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonAuthorized component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#318cbf",
        "borderRadius": 7,
        "padding": 10,
      }
    }
  >
    <IconClass
      brand={false}
      color="#ffffff"
      light={false}
      name="user-shield"
      regular={false}
      size={100}
      solid={false}
    />
  </View>
  <Text
    style={
      Object {
        "color": "#575757",
        "fontSize": 20,
        "fontWeight": "700",
        "marginTop": 5,
      }
    }
  >
    Login first!
  </Text>
  <Text
    style={
      Object {
        "color": "#575757",
        "fontSize": 16,
      }
    }
  >
    Login first to view the screen
  </Text>
  <Button
    containerStyle={
      Object {
        "marginTop": 30,
      }
    }
    onPress={[Function]}
    textStyle={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "textAlign": "center",
        "width": 200,
      }
    }
    title="Login Now"
  />
  <Link
    onPress={[Function]}
    title="New here? Sign Up"
  />
</View>
`;
